loggroup-lambda-connector/src/loggroup-lambda-connector.js

```diff
@@ -23,13 +26,17 @@ function filterLogGroups(event, logGroupRegex) {
     }
 }
 
-function subscribeExistingLogGroups(logGroups) {
+async function subscribeExistingLogGroups(logGroups) {
     var logGroupName;
     var logGroupRegex = new RegExp(process.env.LOG_GROUP_PATTERN, "i");
     var lambdaArn = process.env.LAMBDA_ARN;
     for (var i = logGroups.length - 1; i >= 0; i--) {
         logGroupName = logGroups[i].logGroupName;
         if (logGroupName.match(logGroupRegex)) {
+
+            // sleep time between calls
+            await sleep(1000*(process.env.SUBSCRIBE_DELAY_SECONDS || 2)); // 5 seconds
```

Comment:
@himanshu219 Just One concern, as user has more lambda log groups(Lets say above 400 - 500), are there any chances that the lambda will go in timeout? I mean the 10-15 minutes time out for lambda?

Reply:
Yes it can go to time out.

Reply:
Customers should use more granular regex